app/connectors/connectors/jms/amqp-connector/src/main/java/io/syndesis/connector/amqp/AMQPVerifier.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.connector.amqp;
+
+import java.util.List;
+import java.util.Map;
+
+import io.syndesis.verifier.api.ComponentVerifier;
+import io.syndesis.verifier.api.VerifierResponse;
+
+import org.apache.camel.CamelContext;
+import org.springframework.stereotype.Component;
+
+@Component("amqp")
+public class AMQPVerifier extends ComponentVerifier {
+    public AMQPVerifier() {
+        super("amqp-publish", AMQPVerifierExtension.class);
+    }
+
+    @Override
+    public List<VerifierResponse> verify(CamelContext context, String connectorId, Map<String, Object> params) {
+        // create and register amqp-publish component, if needed
+        if (context.getComponent("amqp-publish") == null) {
+            context.addComponent("amqp-publish",
+                    new AMQPConnectorFactory().newInstance(connectorId, "amqp"));
+        }
+        return super.verify(context, connectorId, params);
+    }
```

Comment:
@dhirajsb I've relaxed access to ComponentVerifier's methods so you do not need this "hack" anymore but you can just implmented `resolveComponentVerifierExtension`, see  https://github.com/syndesisio/syndesis/blob/master/app/connectors/connector-verifier/src/main/java/io/syndesis/verifier/api/ComponentVerifier.java#L93